std/format/internal/floats.d

```diff
@@ -28,13 +28,13 @@ if (is(T == float) || is(T == double)
     return w.data;
 }
 
-package(std.format) void printFloat(Writer, T, Char)(auto ref Writer w, T val, FormatSpec!Char f)
+package(std.format) void printFloat(Writer, T, Char)(auto ref Writer w, const(T) val, FormatSpec!Char f)
 if (is(T == float) || is(T == double)
     || (is(T == real) && (T.mant_dig == double.mant_dig || T.mant_dig == 64)))
 {
     import std.math.operations : extractBitpattern, FloatingPointBitpattern;
 
-    auto bp = extractBitpattern(val);
+    auto bp = extractBitpattern(cast() val);
```

Comment:
make `extractBitpattern` take a const parameter

Reply:
That's gonna require a temporary in  extractBitPattern . Is that ok?

Reply:
Done.